pkg/common-controller/groupsnapshot_controller_helper.go

```diff
@@ -567,14 +567,35 @@ func (ctrl *csiSnapshotCommonController) updateGroupSnapshotStatus(groupSnapshot
 	var pvcVolumeSnapshotRefList []crdv1alpha1.PVCVolumeSnapshotPair
 	if groupSnapshotContent.Status != nil && len(groupSnapshotContent.Status.PVVolumeSnapshotContentList) != 0 {
 		for _, contentRef := range groupSnapshotContent.Status.PVVolumeSnapshotContentList {
-			groupSnapshotContent, err := ctrl.contentLister.Get(contentRef.VolumeSnapshotContentRef.Name)
+			pv, err := ctrl.client.CoreV1().PersistentVolumes().Get(context.TODO(), contentRef.PersistentVolumeRef.Name, metav1.GetOptions{})
+			if err != nil {
+				klog.Errorf(
+					"updateGroupSnapshotStatus[%s]: unable to get PV [%s] from the API server",
+					utils.GroupSnapshotKey(groupSnapshot),
+					contentRef.PersistentVolumeRef.Name)
+			}
+
+			var pvcReference v1.LocalObjectReference
+			if pv != nil {
+				if pv.Spec.ClaimRef != nil {
+					pvcReference.Name = pv.Spec.ClaimRef.Name
+				} else {
+					klog.Errorf(
+						"updateGroupSnapshotStatus[%s]: PV [%s] is not bound",
```

Comment:
As stated above, we should differentiate the case when PV is not found vs PV is not bound.